packages/framework-core/src/services/graphql/graphql-generator.ts

```diff
@@ -198,13 +221,78 @@ export class GraphQLGenerator {
       sortBy: {},
     }
   }
+
+  private static getFields(info: GraphQLResolveInfo): string[] {
+    let fields: string[] = []
+
+    const isList = (type: any): boolean => {
+      if (type instanceof GraphQLNonNull) {
+        return type.ofType instanceof GraphQLList
+      }
+      return type instanceof GraphQLList
+    }
+
+    const extractFields = (
+      selectionSet: SelectionSetNode,
+      path: string[] = [],
+      parentType: GraphQLObjectType | any
+    ): string[] => {
+      let subFields: string[] = []
+      if (selectionSet && selectionSet.selections) {
+        selectionSet.selections.forEach((selection: any) => {
+          const fieldName = selection.name.value
+          const field = parentType.getFields()[fieldName]
+
+          if (!field) {
+            console.error(`Field ${fieldName} not found on type ${parentType.name}. Skipping.`) // @TODO: remove
```

Comment:
Just checking: there is a TODO here that says remove... I don't fully understand this yet, but we are logging an error but saying "skipping".

My question is: Is this truly an error or some condition that can appear and it is ok to continue?

Reply:
@alvaroloes it is not an error; it is expected, actually. It was used for debugging during development to indicate when a field was not part of the selection set. That's why the `TODO` was there but it was removed in commit f498f423. :wink: